sdks/java/managed/src/main/java/org/apache/beam/sdk/managed/Managed.java

```diff
@@ -194,7 +199,28 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
       SchemaTransform underlyingTransform =
           new ManagedSchemaTransformProvider(getSupportedIdentifiers()).from(managedConfig);
 
-      return input.apply(underlyingTransform);
+      return inputTuple.apply(underlyingTransform);
+    }
+
+    @VisibleForTesting
+    static PCollectionRowTuple resolveInput(PInput input) {
```

Comment:
Will it be possible to introduce a similar simplification for output so that callers do not have to call `getOutput` when there's only one output `PCollection` ?

Reply:
Will have to think about this one some more. From what I see, it's possible if ManagedTransform returns a `POutput`, but that may be harmful. POutput doesn't implement `apply()` so it would prevent users from chaining downstream transforms.

We could also create an abstraction between PCollectionRowTuple and PCollection<Row> but that feels a little messy

Reply:
May be we can implement `public <OutputT extends POutput> OutputT apply(PTransform)` for `PCollectionRowTuple` and error out if there's more than on `PCollection` ?

Reply:
PCollectionRowTuple already implements apply() for PTransform<PCollectionRowTuple, OutputT>: 
https://github.com/apache/beam/blob/4ad7037b55f4b711a67f3f2170bc8661a9b799b5/sdks/java/core/src/main/java/org/apache/beam/sdk/values/PCollectionRowTuple.java#L196-L199

Java doesn't let us define the same method with a different generic (e.g. PTransform<PCollection<Row>, OutputT>) because they will ultimately have the same erasure.

Reply:
I think we should think of ways to get rid of the `getSinglePCollection()` call in the future but this is already a very good improvement.